src/Data/AVL/Key/Relation/Order.agda

```diff
@@ -0,0 +1,127 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Pointwise lifting of orders for keys for AVL trees.
+------------------------------------------------------------------------
+
+open import Relation.Binary
+
+module Data.AVL.Key.Relation.Order
+       {k r} (Key : Set k) (_≤_ : Rel Key r)
+       where
+
+open import Data.AVL.Key Key
+
+open import Level
+open import Data.Empty
+open import Data.Unit using (⊤)
+open import Data.Sum
+open import Relation.Nullary
+
+infix 4 _≤⁺_
+
+_≤⁺_ : Key⁺ → Key⁺ → Set r
```

Comment:
The proofs below would be much shorter if you defined this as a datatype?

Reply:
I agree. I stuck to the original style of the module here. But when I had to implement this myself in Idris I did opt for an inductive definition (of the strict order):

https://github.com/gallais/idris-tmustache/blob/master/src/TMustache/Data/TwoThree/Key.idr

If no one is too concerned about the backwards compatibility, I will gladly switch to the inductive definition!

Reply:
Hmm well the problem is that if we want to expose these modules for re-use elsewhere, then we really want to do it the right way as we don't want to encourage this style elsewhere...

Decisions, decisions... I'm a bit worried I'm pushing this "we can break things for v1.0" a bit too far, but maybe this is exactly the sort of thing we should take the opportunity to fix.

Reply:
I don't know how used `AVL` is. Every time I have heard it mentioned it's because people wanted to use `String`s as keys and that is broken (cf. #256).

Note that if we don't want to break backwards compatibility here we could keep `Data.AVL.Key` as it is and simply do our own thing in the `Relation.Binary.Whatever` modules.